lib/src/Uncertainty/Model/DistributionImplementation.cxx

```diff
@@ -2808,6 +2835,219 @@ Graph DistributionImplementation::drawPDF() const
+  if (isDiscrete()) return drawDiscreteLogPDF(xMin, xMax, pointNumber);
+  // Discretization of the x axis
+  const String title(OSS() << getDescription()[0] << " log PDF");
+  NumericalSample gridX;
+  const NumericalSample gridY(computeLogPDF(xMin, xMax, pointNumber, gridX));
+  Curve curvePDF(gridX, gridY);
+  curvePDF.setColor("red");
+  curvePDF.setLegend(title);
+  curvePDF.setLineStyle("solid");
+  curvePDF.setLineWidth(2);
+  const String xName(getDescription()[0]);
+  Graph graphPDF(title, xName, "log PDF", true, "topright");
+  graphPDF.add(curvePDF);
+  NumericalPoint boundingBox(graphPDF.getBoundingBox());
+  const NumericalScalar yMin = boundingBox[2];
+  const NumericalScalar yMax = gridY.computeQuantilePerComponent(0.99)[0];
+  boundingBox[2] = yMin - 0.1 * (yMax - yMin);
+  boundingBox[3] = yMax + 0.1 * (yMax - yMin);
+  graphPDF.setBoundingBox(boundingBox);
+  return graphPDF;
+}
+
+/* Draw the log-PDF of the distribution when its dimension is 1 */
+Graph DistributionImplementation::drawLogPDF(const UnsignedInteger pointNumber) const
+{
+  if (getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: this method is available only for 1D distributions";
+  // For discrete distributions, use the numerical range to define the drawing range
+  const NumericalScalar xMin = computeQuantile(ResourceMap::GetAsNumericalScalar("Distribution-QMin"))[0];
+  const NumericalScalar xMax = computeQuantile(ResourceMap::GetAsNumericalScalar("Distribution-QMax"))[0];
+  const NumericalScalar delta = 2.0 * (xMax - xMin) * (1.0 - 0.5 * (ResourceMap::GetAsNumericalScalar("Distribution-QMax" ) - ResourceMap::GetAsNumericalScalar("Distribution-QMin")));
+  if (isDiscrete())
+  {
+    NumericalScalar a = std::max(xMin - delta, range_.getLowerBound()[0] - 1.0);
+    NumericalScalar b = std::min(xMax + delta, range_.getUpperBound()[0] + 1.0);
+    if (b <= a)
+    {
+      a -= 1.0;
+      b += 1.0;
+    }
+    return drawLogPDF(a, b, pointNumber);
+  }
+  return drawLogPDF(xMin - delta, xMax + delta, pointNumber);
+}
+
+/* Draw the log-PDF of a 1D marginal */
+Graph DistributionImplementation::drawMarginal1DLogPDF(const UnsignedInteger marginalIndex,
+    const NumericalScalar xMin,
+    const NumericalScalar xMax,
+    const UnsignedInteger pointNumber) const
+{
+  Graph marginalGraph(getMarginal(marginalIndex)->drawLogPDF(xMin, xMax, pointNumber));
+  marginalGraph.setTitle(OSS() << getDescription() << "->" << description_[marginalIndex] << " component log PDF");
+  return marginalGraph;
+}
+
+/* Draw the log-PDF of the distribution when its dimension is 2 */
+Graph DistributionImplementation::drawLogPDF(const NumericalPoint & xMin,
+    const NumericalPoint & xMax,
+    const Indices & pointNumber) const
+{
+  if ((pointNumber[0] <= 2) || (pointNumber[1] <= 2)) throw InvalidArgumentException(HERE) << "Error: the discretization must have at least 2 points per component";
```

Comment:
Error message says that pointNumber must be at least 2, but test is wrong, it checks whether it is strictly superior to 2.
BTW please write rewrite tests to handle the case where value is not a number:
```
  if (!(pointNumber[0] >= 2 && pointNumber[1] >= 2)) throw...
```